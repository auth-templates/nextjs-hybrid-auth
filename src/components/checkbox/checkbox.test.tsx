import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Checkbox from './checkbox';

describe('Checkbox', () => {
    test('if all elements are present', () => {
        render(<Checkbox />);
        expect(screen.getByRole('checkbox')).toBeInTheDocument();
        expect(screen.getByTestId('checkbox-cover')).toBeInTheDocument();
    }),

    test('if it is not checked by default', () => {
        render(<Checkbox />);
        expect(screen.getByRole('checkbox')).not.toBeChecked();
    }),

    test('if it changes state properly', () => {
        render(<Checkbox />);
        const checkbox = screen.getByRole('checkbox');
        const checkboxCover = screen.getByTestId('checkbox-cover');
        expect(checkbox).not.toBeChecked();
        userEvent.click(checkboxCover);
        expect(checkbox).toBeChecked();
        userEvent.click(checkboxCover);
        expect(checkbox).not.toBeChecked();
    }),

    it('renders correctly', () => {
        const tree = renderer.create(<Checkbox />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})