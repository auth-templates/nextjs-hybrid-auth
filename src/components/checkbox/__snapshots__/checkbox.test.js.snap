// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="container"
>
  <input
    checked={false}
    id="checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark"
    data-testid="checkbox-cover"
    onClick={[Function]}
  />
</div>
`;
